cmd/secret-manager/util.py

```
@@ -0,0 +1,370 @@
+            f"You don't have permission to access secrets in collection '{collection}'."
+        )
+    except Exception as e:
+        raise click.ClickException(
+            f"Failed to list secrets for collection '{collection}': {e}"
+        )
+
+    secret_list = []
+    try:
+        secret_list = yaml.safe_load(response.payload.data.decode("UTF-8"))
+    except yaml.YAMLError as e:
+        raise click.ClickException(f"Failed to parse the index secret: {e}")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)